releasenotes/notes/rollback_if-039fc1c97caeed37.yaml

```diff
@@ -0,0 +1,22 @@
+---
+features:
+  - |
+    A new pass manager controller `rollback_if` allows to "undo" passes under certain conditions.
+    If the condition is true the passes appended in that set are rolled back. Take for example a
+    pass the processes of choosing the best swapper for a particular situation.
+    The following pass manager applies ``Swapper1`` and calculates the depth of the swapper, saving
+    The result in ``property_set['depth_swapper1']``:
+
+    ```python
+    passmanager = PassManager()
+    passmanager.append([Swapper1(), depth('depth_swapper1')])
+    ```
+
+    Following, the next stage will apply and calculate the depth using ``Swapper2``. If the result is
+    not better, this stage is unrolled, leaving the circuit as it was after applying ``Swapper1`` :
+
+    ```python
+    passmanager.append([Swapper2(), depth('depth_swapper2')],
+                       rollback_if=lambda property_set:
+                       property_set['depth_swapper2'] < property_set['depth_swapper1'])
```

Comment:
I think you have this backwards here. If `depth_swapper2` is smaller, it should not roll back. Lower is better.